app/src/main/java/com/github/se/cyrcle/ui/theme/molecules/TopAppBar.kt

```diff
@@ -49,10 +61,57 @@ fun TopAppBar(navigationActions: NavigationActions, title: String, testTag: Stri
               scrolledContainerColor = MaterialTheme.colorScheme.primaryContainer),
       modifier = Modifier.testTag(testTag),
       navigationIcon = {
-        IconButton(
-            onClick = { navigationActions.goBack() },
-            modifier = Modifier.testTag("${testTag}GoBackButton")) {
-              Icon(Icons.AutoMirrored.Filled.ArrowBack, contentDescription = "Go Back")
-            }
+        Box(modifier = Modifier.padding(8.dp)) {
+          IconButton(
+              onClick = { navigationActions.goBack() },
+              modifier =
+                  Modifier.testTag("${testTag}GoBackButton")
+                      .background(Color.White.copy(alpha = 0.1f), shape = RoundedCornerShape(12.dp))
+                      .size(40.dp)) {
+                Icon(
+                    Icons.AutoMirrored.Filled.ArrowBack,
+                    contentDescription = "Go Back",
+                    modifier = Modifier.size(24.dp))
+              }
+        }
+      })
+}
+
+/**
+ * A composable function that manages the resizing of the text to fit the available space. The text
+ * will be resized until it fits the available space, or until the font size is 6sp. Both the font
+ * size and the line height will be reduced by 5% at each iteration.
+ *
+ * @param text The text to display
+ * @param initialFontSize The initial font size of the text
+ * @param modifier The modifier of the text
+ * @param maxLines The maximum number of lines the text can take
+ */
+@Composable
+fun AutoResizingText(
+    text: String,
+    initialFontSize: TextUnit,
+    modifier: Modifier = Modifier,
+    maxLines: Int
+) {
+  var fontSize by remember { mutableStateOf(initialFontSize) }
+  var lineHeight by remember { mutableStateOf(initialFontSize * 1.2f) }
+  var readyToDraw by remember { mutableStateOf(false) }
+
+  Text(
+      text = text,
+      color = Color.White,
+      fontSize = fontSize,
+      fontWeight = FontWeight.Bold,
+      lineHeight = lineHeight,
+      maxLines = maxLines,
+      modifier = modifier.drawWithContent { if (readyToDraw) drawContent() },
+      onTextLayout = { textLayoutResult ->
+        if (textLayoutResult.didOverflowHeight && fontSize > 6.sp) {
+          fontSize *= 0.95f
+          lineHeight *= 0.95f
```

Comment:
Just for those two values did you consider a constant value for 0.95f if they are both related of course.